services/state/token/token.proto

```diff
@@ -0,0 +1,164 @@
+     */
+    int64 treasury_account_number = 6;
+    /**
+     * (Optional) The admin key of this token. If this key is set, the token is mutable.
+     * A mutable token can be modified.
+     * If this key is not set on token creation, it cannot be modified.
+     */
+    Key admin_key = 7;
+    /**
+     * (Optional) The kyc key of this token.
+     * If this key is not set on token creation, it can only be set if the token has admin key set.
+     */
+    Key kyc_key = 8;
+    /**
+     * (Optional) The freeze key of this token. This key is needed for freezing the token.
+     * If this key is not set on token creation, it can only be set if the token has admin key set.
+     */
+    Key freeze_key = 9;
+    /**
+     * (Optional) The wipe key of this token. This key is needed for wiping the token.
+     * If this key is not set on token creation, it can only be set if the token has admin key set.
+     */
+    Key wipe_key = 10;
+    /**
+     * (Optional) The supply key of this token. This key is needed for minting or burning token.
+     * If this key is not set on token creation, it can only be set if the token has admin key set.
+     */
+    Key supply_key = 11;
+    /**
+     * (Optional) The fee schedule key of this token. This key should be set, in order to make any
+     * changes to the custom fee schedule.
+     * If this key is not set on token creation, it can only be set if the token has admin key set.
+     */
+    Key fee_schedule_key = 12;
+    /**
+     * (Optional) The pause key of this token. This key is needed for pausing the token.
+     * If this key is not set on token creation, it can only be set if the token has admin key set.
+     */
+    Key pause_key = 13;
+    /**
+     * The last used serial number of this token.
+     */
+    int64 last_used_serial_number = 14;
+    /**
+     * The flag indicating if this token is deleted.
+     */
+    bool deleted = 15;
+    /**
+     * The number of the account (if any) that the network will attempt to charge for the
+     * token's auto-renewal upon expiration.
+     */
+    int64 auto_renew_account_number = 16;
+    /**
+     * The number of seconds the network should automatically extend the token's expiration by, if the
+     * token has a valid auto-renew account, and is not deleted upon expiration.
+     * If this is not provided in a allowed range on token creation, the transaction will fail with INVALID_AUTO_RENEWAL_PERIOD.
+     * The default values for the minimum period and maximum period are 30 days and 90 days, respectively.
+     */
+    int64 auto_renew_secs = 17;
+    /**
+     * The expiration time of the token, in seconds since the epoch.
+     */
+    int64 expiry = 18;
+    /**
+     * An optional description of the token with UTF-8 encoding up to 100 bytes.
+     */
+    string memo = 19;
+    /**
+     * The maximum supply of this token.
+     */
+    int64 max_supply = 20;
```

Comment:
Should this be in the first 16 fields?

Reply:
I think all keys should also be in first 16 fields , so the fields are already filled in. 🙂